mypy/checkmember.py

```diff
@@ -1150,7 +1151,8 @@ class B(A[str]): pass
             t = freshen_all_functions_type_vars(t)
             t = bind_self(t, original_type, is_classmethod=True)
             assert isuper is not None
-            t = cast(CallableType, expand_type_by_instance(t, isuper))
+            t = expand_type_by_instance(t, isuper)
```

Comment:
Wonder if we could enhance the type of `expand_type_by_instance` and `expand_type` to make these asserts unnecessary. Seems like it's close to `def expand_type[T: Type](t: T) -> T:`, but maybe it's not quite there.

Reply:
That looks plausible, but I don't really feel confident enough I fully understand that function to make changes there tbh.